sctp.c

```diff
@@ -295,55 +271,60 @@ int janus_sctp_association_setup(janus_sctp_association *sctp) {
 #ifdef HAVE_SCONN_LEN
 	sconn.sconn_len = sizeof(struct sockaddr_conn);
 #endif
-	if(usrsctp_connect(sock, (struct sockaddr *)&sconn, sizeof(struct sockaddr_conn)) < 0) {
+	int res = usrsctp_connect(sctp->sock, (struct sockaddr *)&sconn, sizeof(struct sockaddr_conn));
+	if(res < 0 && errno != EINPROGRESS) {
 		JANUS_LOG(LOG_ERR, "[%"SCNu64"] Error connecting to SCTP server at port %"SCNu16" (%d)\n", sctp->handle_id, sctp->remote_port, errno);
 		return -1;
 	}
 	JANUS_LOG(LOG_VERB, "[%"SCNu64"] Connected to the DataChannel peer\n", sctp->handle_id);
```

Comment:
But if we're ignoring `EINGPROGRESS`, doesn't this mean we're giving the channel for connected, even if it might very well not be?

Reply:
Right, but that could have been the case before, too (e.g. if the setup thread did not run yet or was not finished). The code also tracks the state of the different channels in the SCTP association.